src/components/ButtonDropdown/ButtonDropdown.tsx

```diff
@@ -0,0 +1,213 @@
+import clsx from 'clsx';
+import React, { ReactNode, SyntheticEvent } from 'react';
+import styles from './ButtonDropdown.module.css';
+import { DropdownMenu } from '../..';
+import { ESCAPE_KEYCODE } from '../../util/keycodes';
+import type { ClickableStyleProps } from '../ClickableStyle';
+
+interface FocusEvent<T = Element> extends SyntheticEvent<T> {
+  relatedTarget: EventTarget | null;
+  target: EventTarget & T;
+}
+
+export interface Props {
+  buttonAriaLabel?: string;
+  /**
+   * Makes button full width
+   */
+  fullWidth?: boolean;
+  /**
+   * Adds status to the button (e.g. error, success)
+   */
+  buttonStatus?: ClickableStyleProps<'button'>['status'];
+  /**
+   * Sets button size
+   */
+  buttonSize?: ClickableStyleProps<'button'>['size'];
+  /**
+   * Adds status to the button (e.g. error, success)
+   */
+  buttonVariant?: ClickableStyleProps<'button'>['variant'];
+  /**
+   * Disables the field and prevents editing the contents
+   */
+  disabled?: boolean;
+  /**
+   * Determines type of clickable
+   * - default renders a dropdown menu to the bottom left of the button
+   * - **top-left** renders a dropdown menu to the top left of the button
+   * - **top-right** renders a dropdown menu to the top right of the button
+   * - **bottom-right** renders a dropdown menu to the bottom right of the button
+   */
+  position?: 'top-left' | 'top-right' | 'bottom-right';
+  /**
+   * Determines type of clickable
+   * - **button** The clickable is a clickable button.
+   * - **submit** The clickable is a submit clickable (submits form data).
+   * - **reset** The clickable is a reset clickable (resets the form-data to its initial values)
+   */
+  type?: 'button' | 'reset' | 'submit';
+  /**
+   * Dropdown trigger
+   */
+  dropdownMenuTrigger?: ReactNode;
+  /**
+   * Child node(s) that can be nested inside component
+   */
+  children?: ReactNode;
```

Comment:
Could you add more detail or a usage example to these 2 comments?

Reply:
Yep